### PR TITLE
PLA-4087 - send error messages and notify developers about unknown ac…

### PR DESCRIPTION
**Pull request by kimslava93**
…tions or services.

### CODE REVIEW

File: lib/snsHelper.js [hunk 1]
```diff
@@ -2,30 +2,56 @@ const logger = require('./logger');
...
+            .then((AWS) => {
+                self.SNS = AWS.sns;
+                return StateHelper.getCurrentState();
+            })
+            .then((currentState) => {
+                self.TopicArn = currentState.topicArn;
+            });
+    }
+
+    sendSimpleMessage(message) {
```
Comment (jereynolds) [inline lib/snsHelper.js]: Do these messages not have a type? the engine aggregator is looking for messages with Type: 'Notification'

File: lib/snsHelper.js [hunk 2]
```diff
@@ -2,30 +2,56 @@ const logger = require('./logger');
...
+        }
+        return sendMessageToSnsTopic(this.SNS, this.TopicArn, message);
+    }
+
+    sendTypedMessage(message, messageType) {
+        if (!this.SNS || !this.TopicArn || !message) {
+            throw new Error('No topic ARN or message were found');
+        }
+        const typedMessage = {
+            type: messageType,
```
Comment (jereynolds) [inline lib/snsHelper.js]: engine aggregator is looking for key `Type` capitalized

File: lib/snsHelper.js (hunk 1 above)
Comment (kimslava93) [inline lib/snsHelper.js]: This is custom message structure. So SNS will see that in the 'Message' property.

File: lib/snsHelper.js (hunk 2 above)
Comment (kimslava93) [inline lib/snsHelper.js]: Also custom structure for message, that will be seen only in the 'Message' property

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,5 @@
 node_modules/
+mitmdump.log
 .idea/
-
-*.log
-*.env
```
Comment (kimslava93) [inline .gitignore]: Ok I will remove it

File: lib/snsHelper.js (hunk 2 above)
Comment (jereynolds) [inline lib/snsHelper.js]: sorry i'm still confused how these get converted to types. in the engine https://bitbucket.org/cloudcoreo/engine/src/32b20c4915c3463844dab4b21d414f8ee92b98dc/lib/sqs_aggregator.rb?at=dev&fileviewer=file-view-default#sqs_aggregator.rb-48 we are looking for specific types and will raise an error if they don't match

Comment (kimslava93) [inline lib/snsHelper.js]: It is made for Natalia's PR, where we define structure for SNS message. We get structure of SNS , which consists `Message` property, inside it, we put another field, called `type`. So if structure is not like Aggregator used to accept and work on, we deny it, and notify with this message our developers.

**Pull request closed without merging**